### ISSUE TITLE
[Mobile]Even if the keyboard is present, it will still pop up again

### ISSUE THREAD

**Issue by lcm1475** [singerdmx/flutter-quill#1385]
My issue is about [Mobile]

In the example, only one input box was added
I hope the keyboard won't pop up again

flutter:3.10
flutter-quill:7.3.2
![WX20230830-110332@2x](https://github.com/singerdmx/flutter-quill/assets/9637601/605a2a00-b7ea-49dc-a6da-529e3d0e1270)

https://github.com/singerdmx/flutter-quill/assets/9637601/c6fd7cb2-004b-4800-bfe2-713daf910b3d

**Comment by EgorK0rshun** [comment]
`skipRequestKeyboard = true` property for quillController?

**Comment by lcm1475** [comment]
I tried, but I can't

**Comment by lcm1475** [comment]
@singerdmx

**Comment by WillianSalceda** [comment]
That's definitely a problem.

Using the same code as the issue presented here, when switching from TextField to QuillEditor, the keyboard inputs are not "computed". You have to press again in the QuillEditor before you can type anything.
The keyboard receives the input, so much so that text based on what has been written appears in the suggestions, but nothing appears in the app's editor.

The *skipRequestKeyboard* parameter doesn't solve the problem either.

When removing all calls to open the keyboard within the package and calling it through the app, this problem is solved (which is not ideal, and also triggers new issues).
In other words, it actually happens because the keyboard is opened again, even if it was already opened before due to another TextField.

**Comment by verhneudinec** [comment]
any updates?
@singerdmx

**Comment by ReptileStyle** [comment]
i made some garbage fixes that worked for me in slightly different case. But yes, i got multiple TextFields and when i focus QuillEditor, i can see keyboard, but nothing printed in QuillEditor. Note: i am using quill version 7.4.7 because i am not able to migrate to new flutter currently.


`onTapUp: (details, updatePosition) {
                          // при фокусе в поле нет никаких проблем
                          if (widget.noteFocus.hasFocus) {
                            return false;
                          }
                          print('requesting focus on tapup');
                          // хак. пока не пофиксили https://github.com/singerdmx/flutter-quill/issues/1385
                          widget.noteFocus.requestFocus();
                          final offset = updatePosition(details.globalPosition).offset;
                          Future.microtask(() async {
                            while(true) {
                              try {
                                widget.noteController.moveCursorToPosition(offset);

                                break;
                              } catch(e) {
                                // retry
                                await Future.delayed(Duration(milliseconds: 10));
                              }
                            }
                          });


                          return true;
                        },`
                        
This fix alone can made it work. But i also have AbsorbPointer on top of me QuillEditor to just expand it on first touch. Maybe problem is not in the AbsorbPointer, because a have more complex code. But here is my fix: add this code in Quill fork

new file. I just reused this code to unsure widget is present in tree. And if not -> await it
`
import 'dart:async';

import 'package:flutter/material.dart';

extension GlobalKeyExtension on GlobalKey {
  /// await widget to be mounted, then returns size
  Future<Size?> getWidgetSize() async {
    // Completer that will complete when the size is available
    final Completer<Size?> completer = Completer<Size?>();

    WidgetsBinding.instance.addPostFrameCallback((_) {
      // Check if the current context is available
      if (currentContext != null) {
        // Add a post-frame callback to ensure the build process is complete
          // After the build is complete, we can get the render box
          final RenderBox? renderBox = currentContext?.findRenderObject() as RenderBox?;
          // If the render box is available, get the size
          final Size? size = renderBox?.size;
          // Complete the completer with the size
          completer.complete(size);

      } else {
        // If the context is not available, complete with null
        completer.complete(null);
      }
    });

    // Return the future of the completer
    return completer.future;
  }

  /// await widget to be mounted, then returns size
  Future<Size> getWidgetSizeForce({int framesLimit = 120}) async {
    for(var i = 0; i < 120; i++) {
      final size = await getWidgetSize();

      if (size != null) {
        return size;
      }
    }

    throw Exception('Widget with key $this was not found in the tree for $framesLimit frames, ${StackTrace.current}');
  }
}`

Find method requestKeyboard in raw_editor.dart and wrap openConnectionIfNeeded() like this

`_editorKey.getWidgetSizeForce().then((value) {
        openConnectionIfNeeded();
        if (!keyboardAlreadyShown) {
          /// delay 500 milliseconds for waiting keyboard show up
          Future.delayed(const Duration(milliseconds: 500), _showCaretOnScreen);
        } else {
          _showCaretOnScreen();
        }
      });`

**Comment by EchoEllet** [comment]
> Note: i am using quill version 7.4.7 because i am not able to migrate to new flutter currently.

What is blocking the migration? If it's a dependency conflict issue, then it might be solvable.
